### PR TITLE
Feature/deployments to auto scaling groups

### PR DESCRIPTION
**Pull request by marko-baric-travcorp**
We added support for deployments to whole auto scaling groups, not only to EC2 instaces directly.

This is needed if we want the ASG to deploy the code to every new instance (e.g. when scaling out) - then Code Deploy must be set up to deploy to ASG and not directly to instances.

### CODE REVIEW

File: TTC.Deployment.Tests/credentials [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[dev]
```
Comment (PaulCampbell) [inline TTC.Deployment.Tests/credentials]: ummm

**Comment by PaulCampbell** [comment]
this is a bit confusing. there seems to be a lot more going on in here than the PR description suggests? Am I missing something?

File: TTC.Deployment.Tests/credentials (hunk 1 above)
Comment (marko-baric-travcorp) [inline TTC.Deployment.Tests/credentials]: This file is removed in another branch. No credentails are stored in the repo.

**Comment by marko-baric-travcorp** [comment]
That's because i had to merge another branch to this one. So you see both changes here. My mistake because I work on three parallel branches.
If you want to come to my desk I can explain what was implemented.

**Comment by PaulCampbell** [comment]
Ok, maybe once the other branch is merged we can rebase this with master and it will make more sense

**Pull request closed without merging**